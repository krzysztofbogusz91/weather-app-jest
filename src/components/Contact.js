import React, { Component } from 'react';
import PropTypes from 'prop-types';


const propTypes = {

};


class Contact extends Component {
    render() {
        return (
            <div className="d-flex flex-column justify-content-center align-content-center m-5">
                <h2 className="align-self-center p-3">Contactc Forms etc.</h2>
                <div>
                <form>
                    <div class="form-group">
                        <label for="email">Email address</label>
                        <input type="email" class="form-control" id="email" aria-describedby="emailHelp" placeholder="Enter email"/>
                        </div>
                        <div class="form-group">
                            <label for="text-area">Messsage</label>
                            <textarea type="text" class="form-control" id="text-area" />
                        </div>
                        </form>
                        </div>
                    </div>
                    );
                }
            }
            
            
            Contact.propTypes = propTypes;
            
            
export default Contact;