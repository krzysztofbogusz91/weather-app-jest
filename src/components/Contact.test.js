import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import set from '../enzyme_set_up';
import Contact from './Contact';

describe('User',()=>{
    const con = shallow( <Contact /> )

    it('renders without crashing', () => {
        expect(con).toMatchSnapshot();
    });
      
})
