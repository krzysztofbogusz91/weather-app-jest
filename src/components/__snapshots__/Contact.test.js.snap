// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Contact />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="align-self-center p-3"
>
          Contactc Forms etc.
</h2>,
        <div>
          <form>
                    <div
                              class="form-group"
                    >
                              <label
                                        for="email"
                              >
                                        Email address
                              </label>
                              <input
                                        aria-describedby="emailHelp"
                                        class="form-control"
                                        id="email"
                                        placeholder="Enter email"
                                        type="email"
                              />
                    </div>
                    <div
                              class="form-group"
                    >
                              <label
                                        for="text-area"
                              >
                                        Messsage
                              </label>
                              <textarea
                                        class="form-control"
                                        id="text-area"
                                        type="text"
                              />
                    </div>
          </form>
</div>,
      ],
      "className": "d-flex flex-column justify-content-center align-content-center m-5",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Contactc Forms etc.",
          "className": "align-self-center p-3",
        },
        "ref": null,
        "rendered": "Contactc Forms etc.",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form>
            <div
                        class="form-group"
            >
                        <label
                                    for="email"
                        >
                                    Email address
                        </label>
                        <input
                                    aria-describedby="emailHelp"
                                    class="form-control"
                                    id="email"
                                    placeholder="Enter email"
                                    type="email"
                        />
            </div>
            <div
                        class="form-group"
            >
                        <label
                                    for="text-area"
                        >
                                    Messsage
                        </label>
                        <textarea
                                    class="form-control"
                                    id="text-area"
                                    type="text"
                        />
            </div>
</form>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                class="form-group"
>
                <label
                                for="email"
                >
                                Email address
                </label>
                <input
                                aria-describedby="emailHelp"
                                class="form-control"
                                id="email"
                                placeholder="Enter email"
                                type="email"
                />
</div>,
              <div
                class="form-group"
>
                <label
                                for="text-area"
                >
                                Messsage
                </label>
                <textarea
                                class="form-control"
                                id="text-area"
                                type="text"
                />
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label
                    for="email"
>
                    Email address
</label>,
                  <input
                    aria-describedby="emailHelp"
                    class="form-control"
                    id="email"
                    placeholder="Enter email"
                    type="email"
/>,
                ],
                "class": "form-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Email address",
                    "for": "email",
                  },
                  "ref": null,
                  "rendered": "Email address",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-describedby": "emailHelp",
                    "class": "form-control",
                    "id": "email",
                    "placeholder": "Enter email",
                    "type": "email",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label
                    for="text-area"
>
                    Messsage
</label>,
                  <textarea
                    class="form-control"
                    id="text-area"
                    type="text"
/>,
                ],
                "class": "form-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Messsage",
                    "for": "text-area",
                  },
                  "ref": null,
                  "rendered": "Messsage",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "class": "form-control",
                    "id": "text-area",
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "textarea",
                },
              ],
              "type": "div",
            },
          ],
          "type": "form",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="align-self-center p-3"
>
            Contactc Forms etc.
</h2>,
          <div>
            <form>
                        <div
                                    class="form-group"
                        >
                                    <label
                                                for="email"
                                    >
                                                Email address
                                    </label>
                                    <input
                                                aria-describedby="emailHelp"
                                                class="form-control"
                                                id="email"
                                                placeholder="Enter email"
                                                type="email"
                                    />
                        </div>
                        <div
                                    class="form-group"
                        >
                                    <label
                                                for="text-area"
                                    >
                                                Messsage
                                    </label>
                                    <textarea
                                                class="form-control"
                                                id="text-area"
                                                type="text"
                                    />
                        </div>
            </form>
</div>,
        ],
        "className": "d-flex flex-column justify-content-center align-content-center m-5",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Contactc Forms etc.",
            "className": "align-self-center p-3",
          },
          "ref": null,
          "rendered": "Contactc Forms etc.",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <form>
              <div
                            class="form-group"
              >
                            <label
                                          for="email"
                            >
                                          Email address
                            </label>
                            <input
                                          aria-describedby="emailHelp"
                                          class="form-control"
                                          id="email"
                                          placeholder="Enter email"
                                          type="email"
                            />
              </div>
              <div
                            class="form-group"
              >
                            <label
                                          for="text-area"
                            >
                                          Messsage
                            </label>
                            <textarea
                                          class="form-control"
                                          id="text-area"
                                          type="text"
                            />
              </div>
</form>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  class="form-group"
>
                  <label
                                    for="email"
                  >
                                    Email address
                  </label>
                  <input
                                    aria-describedby="emailHelp"
                                    class="form-control"
                                    id="email"
                                    placeholder="Enter email"
                                    type="email"
                  />
</div>,
                <div
                  class="form-group"
>
                  <label
                                    for="text-area"
                  >
                                    Messsage
                  </label>
                  <textarea
                                    class="form-control"
                                    id="text-area"
                                    type="text"
                  />
</div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label
                      for="email"
>
                      Email address
</label>,
                    <input
                      aria-describedby="emailHelp"
                      class="form-control"
                      id="email"
                      placeholder="Enter email"
                      type="email"
/>,
                  ],
                  "class": "form-group",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Email address",
                      "for": "email",
                    },
                    "ref": null,
                    "rendered": "Email address",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-describedby": "emailHelp",
                      "class": "form-control",
                      "id": "email",
                      "placeholder": "Enter email",
                      "type": "email",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label
                      for="text-area"
>
                      Messsage
</label>,
                    <textarea
                      class="form-control"
                      id="text-area"
                      type="text"
/>,
                  ],
                  "class": "form-group",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Messsage",
                      "for": "text-area",
                    },
                    "ref": null,
                    "rendered": "Messsage",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "class": "form-control",
                      "id": "text-area",
                      "type": "text",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "textarea",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "form",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
